src/deploy/functions/release/planner.ts

```diff
@@ -88,11 +112,13 @@ export function createDeploymentPlan(
 
   const regions = new Set([...Object.keys(want.endpoints), ...Object.keys(have.endpoints)]);
   for (const region of regions) {
-    deployment[region] = calculateRegionalChanges(
+    const changesets = calculateChangesets(
       want.endpoints[region] || {},
       have.endpoints[region] || {},
+      (e) => `${e.region}-${e.availableMemoryMb || "default"}`,
```

Comment:
I don't think we actually do heap tuning for all memory sizes. It might be worth talking to the cloud folks to see if we can merge deployments of <2GB

Reply:
Looked in and discovered that the environment variable is set per memory configuration:

https://github.com/GoogleCloudPlatform/buildpacks/blob/663180f2cf1fbfdee4a889f6c9d17ee2eb9091bf/cmd/nodejs/functions_framework/main.go#L166

Seems like they took the simplest path forward.